File: plugins/textmate/src/org/jetbrains/plugins/textmate/language/syntax/selector/TextMateSelectorToken.java
package org.jetbrains.plugins.textmate.language.syntax.selector;

public interface TextMateSelectorToken {
  TextMateSelectorToken COMMA = new TextMateSelectorLexer.SignToken(',');
  TextMateSelectorToken LPAREN = new TextMateSelectorLexer.SignToken('(');
  TextMateSelectorToken RPAREN = new TextMateSelectorLexer.SignToken(')');
  TextMateSelectorToken PIPE = new TextMateSelectorLexer.SignToken('|');
  TextMateSelectorToken MINUS = new TextMateSelectorLexer.SignToken('-');
  TextMateSelectorToken HAT = new TextMateSelectorLexer.SignToken('^');
}

File: plugins/textmate/core/src/org/jetbrains/plugins/textmate/language/syntax/selector/TextMateSelectorToken.kt
package org.jetbrains.plugins.textmate.language.syntax.selector

sealed class TextMateSelectorToken {
  companion object {
    val COMMA: TextMateSelectorToken = SignToken(',')
    val LPAREN: TextMateSelectorToken = SignToken('(')
    val RPAREN: TextMateSelectorToken = SignToken(')')
    val PIPE: TextMateSelectorToken = SignToken('|')
    val MINUS: TextMateSelectorToken = SignToken('-')
    val HAT: TextMateSelectorToken = SignToken('^')
  }

  data class SignToken(val sign: Char) : TextMateSelectorToken()
  data class PriorityToken(val priority: TextMateWeigh.Priority) : TextMateSelectorToken()
  data class SelectorToken(val text: String) : TextMateSelectorToken()
}
